libraries/AP_RCProtocol/AP_CRSF_Out.cpp

```diff
@@ -0,0 +1,388 @@
+            _baud_negotiation_result = BaudNegotiationResult::PENDING;
+            _state = State::WAITING_FOR_RC_LOCK;
+        }
+        break;
+    }
+}
+
+bool AP_CRSF_Out::decode_crsf_packet(AP_CRSF_Protocol::Frame& _frame)
+{
+#ifdef CRSF_RCOUT_DEBUG_FRAME
+    hal.console->printf("CRSFOut: received %s:", AP_CRSF_Protocol::get_frame_type(_frame.type));
+    uint8_t* fptr = (uint8_t*)&_frame;
+    for (uint8_t i = 0; i < _frame.length + 2; i++) {
+        hal.console->printf(" 0x%x", fptr[i]);
+    }
+    hal.console->printf("\n");
+#endif
+
+    switch (_frame.type) {
+        case AP_CRSF_Protocol::FrameType::CRSF_FRAMETYPE_COMMAND: {
+            const AP_CRSF_Protocol::CommandFrame* cmd = (const AP_CRSF_Protocol::CommandFrame*)_frame.payload;
+            if (cmd->origin == AP_CRSF_Protocol::DeviceAddress::CRSF_ADDRESS_FLIGHT_CONTROLLER &&
+                cmd->command_id == AP_CRSF_Protocol::CRSF_COMMAND_GENERAL &&
+                cmd->payload[0] == AP_CRSF_Protocol::CRSF_COMMAND_GENERAL_CRSF_SPEED_RESPONSE) {
+                const bool success = cmd->payload[2];
+                if (success) {
+                    debug_rcout("CRSF Speed Response Received: SUCCESS");
+                    _baud_negotiation_result = BaudNegotiationResult::SUCCESS;
+                    // change baud on our end now
+                    _crsf_port->change_baud_rate(_target_baudrate);
+                } else {
+                    debug_rcout("CRSF Speed Response Received: FAILED");
+                    _baud_negotiation_result = BaudNegotiationResult::FAILED;
+                }
+            }
+        }
+            break;
+        case AP_CRSF_Protocol::FrameType::CRSF_FRAMETYPE_PARAM_DEVICE_INFO:
+            AP_CRSF_Protocol::process_device_info_frame((AP_CRSF_Protocol::ParameterDeviceInfoFrame*)_frame.payload,
+                                                         &version, true);
+            break;
+
+        case AP_CRSF_Protocol::FrameType::CRSF_FRAMETYPE_PARAM_DEVICE_PING:
+            send_device_info();
+            break;
+
+        default:
+            break;
+    }
+
+    return true;
+}
+
+// sends RC frames at the configured rate
+void AP_CRSF_Out::send_rc_frame()
+{
+    uint16_t channels[CRSF_MAX_CHANNELS] {};
+    const uint8_t nchan = MIN(NUM_SERVO_CHANNELS, (uint8_t)CRSF_MAX_CHANNELS);
+
+    for (uint8_t i = 0; i < nchan; ++i) {
+        SRV_Channel *c = SRV_Channels::srv_channel(i);
+        if (c != nullptr) {
+            channels[i] = c->get_output_pwm();
+        } else {
+            channels[i] = 1500; // Default to neutral if channel is null
+        }
+    }
+
+#ifdef CRSF_RCOUT_DEBUG
+    const uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - last_update_debug_ms > 1000) {
+        last_update_debug_ms = now_ms;
+        debug_rcout("Updating channels. CH1=%u CH2=%u CH3=%u", channels[0], channels[1], channels[2]);
+    }
+#endif
+    AP_CRSF_Protocol::Frame frame;
```

Comment:
Either zero this or (more correctly declare it `const` and fill in all of the fields with initialisers).

In either case, don't make it looks like we might use uninitialised stack data.